### PR TITLE
py-contourpy: add v1.1.1

### PR DESCRIPTION
**Pull request by manuelakuhn**
https://github.com/contourpy/contourpy/tree/v1.1.1

for `python@3.12:` this depends on #40057

### CODE REVIEW

**Review by tldahlgren — COMMENTED** [review]
FYI. Confirmed the version sha256.

**Review by adamjstewart — COMMENTED** [review]
See pandas PR, I don't think we actually need to specify what versions of numpy support Python 3.12.

File: var/spack/repos/builtin/packages/py-contourpy/package.py [hunk 1]
```diff
@@ -12,13 +12,20 @@ class PyContourpy(PythonPackage):
     homepage = "https://github.com/contourpy/contourpy"
     pypi = "contourpy/contourpy-1.0.5.tar.gz"
 
+    version("1.1.1", sha256="96ba37c2e24b7212a77da85004c38e7c4d155d3e72a45eeaf22c1f03f607e8ab")
```
Comment (wdconinc) [inline var/spack/repos/builtin/packages/py-contourpy/package.py:17]:
```suggestion
    version("1.2.1", sha256="4d8908b3bee1c889e547867ca4cdc54e5ab6be6d3e078556814a22457f49423c")
    version("1.1.1", sha256="96ba37c2e24b7212a77da85004c38e7c4d155d3e72a45eeaf22c1f03f607e8ab")
```

File: var/spack/repos/builtin/packages/py-contourpy/package.py [hunk 2]
```diff
@@ -12,13 +12,20 @@ class PyContourpy(PythonPackage):
...
     version("1.0.7", sha256="d8165a088d31798b59e91117d1f5fc3df8168d8b48c4acc10fc0df0d0bdbcc5e")
     version("1.0.5", sha256="896631cd40222aef3697e4e51177d14c3709fda49d30983269d584f034acc8a4")
 
     depends_on("python@3.8:", when="@1.0.7:", type=("build", "link", "run"))
```
Comment (wdconinc) [inline var/spack/repos/builtin/packages/py-contourpy/package.py:21]:
```suggestion
    depends_on("python@3.9:", when="@1.2:", type=("build", "link", "run"))
    depends_on("python@3.8:", when="@1.0.7:", type=("build", "link", "run"))
```

File: var/spack/repos/builtin/packages/py-contourpy/package.py [hunk 3]
```diff
@@ -12,13 +12,20 @@ class PyContourpy(PythonPackage):
...
-    depends_on("py-pybind11@2.6:", type=("build", "link"))
-    depends_on("py-setuptools@42:", type="build")
-    depends_on("py-numpy@1.16:", type=("build", "run"))
+    depends_on("meson@1.2:", when="@1.1.1:", type=("build"))
+    depends_on("py-meson-python@0.13.1:", when="@1.1:", type=("build"))
+    depends_on("py-pybind11@2.10.4:", when="@1.1:", type=("build", "link"))
+    depends_on("py-pybind11@2.6:", when="@:1.0", type=("build", "link"))
 
+    depends_on("py-numpy@1.26:1", when="^python@3.12:", type=("build", "run"))
+    depends_on("py-numpy@1.16:1", when="^python@:3.11", type=("build", "run"))
```
Comment (wdconinc) [inline var/spack/repos/builtin/packages/py-contourpy/package.py:29]:
```suggestion
    depends_on("py-numpy@1.16:1", when="^python@:3.11", type=("build", "run"))
    depends_on("py-numpy@1.20:1", when="^python@:3.11", type=("build", "run"), when="@1.2")
```
In main the upper limit has already been removed, so I expect 1.3 to be depend on `py-numpy@1.20:` including allowing for py-numpy-2.

**Review by wdconinc — COMMENTED** [review]
Some comments to add py-contourpy-1.2.1, if you want to expand scope here. If not, also ok and I'll add it when this is merged.

**Comment by wdconinc** [comment]
Merged updates from develop, resolved conflict due to license addition, marked as checked.

**Comment by alalazo** [comment]
Superseded by #46225

**Pull request closed without merging**